Character Animations

jidaprano opened the pull request:
Added all animations, tried to get the timing and transitions OK, weight painted the hoodie.

Play around with it in the game, let me know if anything should be sped up, slowed down, or otherwise refined. 

For awareness, added a root bone

prepsure reviewed (changes requested):
- character appears to move forward from the root position when falling/jumping/long jumping/plunging and when they land it snaps them backward
- the transition to plunging/gliding is very slow and feels sort of unnatural
- roll is probably 3-5 times fast than it should be
- the transition from jumping -> falling is pretty obvious, and because jumping has a pretty much fixed time, it may be good to lead into the fall

things that are probably related to bugs in the movement states, but i'm unsure
- sometimes the character gets stuck in the plunge animation while high jumping, not sure what that's about
- sometimes the character transitions between highjumping and falling in a loop

jidaprano commented:
- Fixed the character moving from root position for mentioned states
- Fixed roll speed
- Fixed the super slow transition to gliding
- Changed the falling animation to have the legs in a similar position to the jump animation to make it feel a bit more smooth
Wasn't really able to decipher solutions to the movement state issues, if they persist after your movement state bugfixes I can investigate further.
But test it out and lmk what you think
